Adding parts of cdn-in-a-box

ocket8888 opened the pull request:
I've added a couple components to cdn-in-a-box, specifically some default configurations, an origin server and a mid server.

asfgit commented:
Can one of the admins verify this patch?

dangogh commented on infrastructure/cdn-in-a-box/mid/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+
+FROM centos:7
+
+ADD https://ci.trafficserver.apache.org/RPMS/CentOS7/trafficserver-7.1.2-4.el7.centos.x86_64.rpm /trafficserver-7.1.2-4.el7.centos.x86_64.rpm
```
this should allow overriding the rpm file location/version for more flexibility

dangogh commented on infrastructure/cdn-in-a-box/mid/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+ADD mid_profile.json /mid_profile.json
+ADD server.json /server.json
+ADD setup.sh /setup.sh
+
```
docker creates a new layer for each line here,  and a lot of repetition.  Combine the mkdir's, chown's, ADD's to reduce that.

dangogh commented on infrastructure/cdn-in-a-box/mid/setup.sh:
```diff
@@ -0,0 +1,101 @@
+done
+
+# Now network things. First need to authenticate
+curl -ksc cookie.jar -d '{"u":"admin","p":"twelve"}' https://trafficportal/api/1.3/user/login
```
hard-coded user/passwd?   you need the same one as stored in traffic_ops -- use values from `variables.env`

dangogh commented on infrastructure/cdn-in-a-box/traffic_ops/Dockerfile-go:
```diff
@@ -25,11 +25,16 @@ FROM centos:7
+ADD cdns.json /
+ADD deliveryservices.json /
+ADD cachegroup.json /
+ADD phys_location.json /
```
we need a more flexible way to do this or it's gonna get unwieldy

dangogh commented on infrastructure/cdn-in-a-box/traffic_ops/setup.sh:
```diff
@@ -0,0 +1,55 @@
+
+# First, we need to authenticate
+curl -skc cookie.jar -d "{\"u\":\"$TO_ADMIN_USER\",\"p\":\"twelve\"}" https://localhost:6443/api/1.3/user/login
+echo
```
it's not `localhost:6443` -- it's `https://$TO_HOST:$TO_PORT`

ocket8888 replied:
But surely since this is running _on_ the traffic_ops machine `https://localhost:$TO_PORT` would be fine?

ocket8888 commented:
I think my latest push addresses most of these issues; however I've left [`traffic_ops/setup.sh`](infrastructure/cdn-in-a-box/traffic_ops/setup.sh) querying the locally-hosted TO instance as `https://localhost:$TO_PORT`

ocket8888 commented on infrastructure/cdn-in-a-box/traffic_ops/setup.sh:
```diff
@@ -0,0 +1,55 @@
+curl -skb cookie.jar -d @/deliveryservices.json https://localhost:$TO_PORT/api/1.3/deliveryservices
+echo
+#... and a cachegroup
+MID_LOC_ID=$(curl -skb cookie.jar https://localhost:$TO_PORT/api/1.3/types | sed -re 's/\},\{/\n/g' | grep MID_LOC | tr ',' '\n' | grep '"id"' | cut -d : -f2)
```
idk what `jq` is, but I'll take your word for it that this works

ocket8888 commented:
I converted everything to use jq pipelines instead of bash pipelines

dewrich commented:
I like the jq pipelines, much cleaner!
